DataFormats/SiStripCluster/src/SiStripApproximateCluster.cc

```diff
@@ -1,15 +1,18 @@
 #include "DataFormats/SiStripCluster/interface/SiStripApproximateCluster.h"
 #include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
 #include <algorithm>
+#include <cassert>
 #include <cmath>
 
 SiStripApproximateCluster::SiStripApproximateCluster(const SiStripCluster& cluster,
                                                      unsigned int maxNSat,
                                                      float hitPredPos,
                                                      bool peakFilter) {
-  barycenter_ = std::round(cluster.barycenter() * 10);
+  compBarycenter_ = std::round(cluster.barycenter() * maxRange_ / maxBarycenter_);
+  assert(cluster.barycenter() <= maxBarycenter_ && "Got a barycenter > maxBarycenter");
+  assert(compBarycenter_ <= maxRange_ && "Filling compBarycenter > maxRange");
```

Comment:
Isn't this condition always true? (maximum possible value of `compBaryCenter_` being `2**16-1 = 65535` and `maxRange_` being set to `65535.`)